File: src/main/kotlin/com/github/hzqd/tank/war/model/Blast.kt
package com.github.hzqd.tank.war.model

import com.github.hzqd.tank.war.Config
import com.github.hzqd.tank.war.business.Destroyable
import org.itheima.kotlin.game.core.Painter

/**爆炸物*/
class Blast(override val x: Int, override val y: Int) : Destroyable {

    override val width: Int = Config.block
    override val height: Int = Config.block
    private val imagePaths = arrayListOf<String>()
    private var index = 0

    init { (1..8).forEach { imagePaths.add("img/blast${it}.png") } }

    override fun draw() {
        val i = index % imagePaths.size
        Painter.drawImage(imagePaths[i], x, y)
        index++
    }

    override fun isDestroyed() = index >= imagePaths.size
}

File: src/main/scala/com/github/hzqd/tank/war/model/Blast.scala
package com.github.hzqd.tank.war.model

import com.github.hzqd.tank.war.Config
import com.github.hzqd.tank.war.business.Destroyable
import org.itheima.kotlin.game.core.Painter.INSTANCE.drawImage
import scala.collection.mutable.ArrayBuffer
import com.github.hzqd.tank.war.ext.Fn.KtStd

/**爆炸物*/
case class Blast(override var x: Int, override var y: Int) extends Destroyable {

    override var width: Int = Config.block
    override var height: Int = Config.block
    private val imagePaths = ArrayBuffer[String]()
    private var index = 0

    (1 to 8).foreach(i => imagePaths.addOne(s"img/blast$i.png"))

    override def draw() {
        (index % imagePaths.size).let(i => drawImage(imagePaths(i), x, y))
        index += 1
    }

    override def isDestroyed() = index >= imagePaths.size
}
